### PR TITLE
Integrate the swift compiler as a built-in tool

### PR DESCRIPTION
**Pull request by hartbit**
Please don't hesitate to code review this in detail, my C++ is *very* rusty.

### CODE REVIEW

File: include/llbuild/BuildSystem/SwiftTools.h [hunk 1]
```diff
@@ -17,14 +17,26 @@
...
+#include "llbuild/BuildSystem/BuildDescription.h"
+
 #include <memory>
 
 namespace llbuild {
 namespace buildsystem {
 
-class Tool;
-
-std::unique_ptr<Tool> createSwiftCompilerTool(StringRef name);
```
Comment (ddunbar) [inline include/llbuild/BuildSystem/SwiftTools.h]: If we are moving this to be a builtin tool, I think we can drop this header completely and just define this tool in BuildSystem.cpp, as we do the other builtin ones.

**Comment by hartbit** [comment]
@ddunbar I've done the required change

**Comment by ddunbar** [comment]
@swift-ci please smoke test

**Comment by hartbit** [comment]
@swift-ci please smoke test

**Comment by hartbit** [comment]
@swift-ci please smoke test

**Comment by hartbit** [comment]
@swift-ci please smoke test

**Comment by hartbit** [comment]
@swift-ci please smoke test

**Comment by hartbit** [comment]
@swift-ci please smoke test

**Pull request merged**